topics/tutorial-first-steps-with-kotlin-rpc.topic

```diff
@@ -0,0 +1,600 @@
+        <p>
+            Let’s try to create an application for ordering pizza over the network. In order to keep the code as simple
+            as possible, we’ll use a console-based client. Future articles will show how it would work with KMP-based
+            desktop and mobile clients.
+        </p>
+        <chapter title="Create the projects" id="create-projects">
+            <p>First, you will create two projects, one for the client and one for the server.</p>
+            <p>
+                It would be possible to create a single project that combines both tasks, but then it becomes hard to
+                appreciate what is really going on. For example, your build file would include both client and
+                server-side dependencies, obscuring what is essential on each side of the network.
+            </p>
+            <procedure id="create-projects-procedure">
+                <step>
+                    Launch <a href="">intelliJ IDEA</a>.
+                </step>
+                <step>
+                    <include from="lib.topic" element-id="new_project_idea"/>
+                </step>
+                <step>
+                    In the
+                    <control>Name</control>
+                    field, enter
+                    <path>KotlinRpcPizzaServer</path>
+                    as the name of your project.
+                    <img src="tutorial_kotlin_rpc_intellij_project.png" alt="IntelliJ New Kotlin Project window"
+                         style="block" width="706" border-effect="rounded"/>
+                </step>
+                <step>
+                    Leave the rest of the default settings and click
+                    <control>Create</control>
+                    .
+                </step>
+                <step>
+                    Repeat the process to create a second project with the name
+                    <path>KotlinRpcPizzaClient</path>
+                    .
+                </step>
+            </procedure>
+            <p>
+                Normally, you would immediately configure the build file of each project. However, that’s an
+                implementation detail that won't enhance your understanding of the technology, so you'll get back to
+                that step at the end.
+            </p>
+        </chapter>
+        <chapter title="Add the shared types" id="shared-types">
+            <p>
+                The heart of any RPC project is the interface that defines the procedures to be called remotely, along
+                with the types used in the definition of those procedures.
+            </p>
+            <p>
+                These types will need to be placed in both the client and server projects. Ideally, you want a build
+                process that packages them into a separate library, which can be shared as needed. However, for
+                simplicity of this example, you can copy and paste.
```

Comment:
> However, for simplicity of this example, you can copy and paste.

I would disagree, that it is simpler. While it may be true from the build setup point of view, the two other aspects are hurt:
- This approach hides the biggest advantage of the Kotlin RPC - write once, use everywhere, 
- It instead uses copy-paste thing, which is generally considered bad and error-prone. I don't we should endorse such an approach even in a toy example project.

I see why it is easier in the example. Still, I think that it would hurt the reader more in the long run. Besides, we already have two modules, adding one more it not that big of a deal, isn't it?

Reply:
Can you please explain what would be the correct procedure if we were to work with a single project from the beginning?
I do agree with you 100% that we should show the proper way of structuring. 
What if we keep this initial approach and add another iteration before the "Extend the example" section where we combine the modules under one project and create a shared one? In this way the reader can really feel the added value of single-sourcing.

Reply:
I think the absolutely best approach would be to have a Ktor Generator sample.

But since there is not one now, I think we should explain, that in one project you can have more than 1 module, exactly for such purposes. And we can even write `now create 3 identical folders: server, client and shared - and place this sample build.gradle.kts in all of them while all still in the same project`, but in more professional style. And the code snippet for these files would not be exactly big or complicated. And we should show the directories structure. Alternatively, same steps may be done from already exiting IDEA template singe-module project, where user just adds two more modules.

I strongly oppose the idea of creating 2 separate projects with copy-pasted services, as it is exactly what you wouldn't ever do with such setup. I agree, that is it way simpler to explain, but I consider it not that good in the long run